src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/controllers/RepoController.java

```diff
@@ -44,49 +44,43 @@
 import uk.ac.cam.cl.dtg.teaching.pottery.task.Task;
 import uk.ac.cam.cl.dtg.teaching.pottery.task.TaskCopy;
 import uk.ac.cam.cl.dtg.teaching.pottery.task.TaskIndex;
+import uk.ac.cam.cl.dtg.teaching.pottery.worker.Job;
+import uk.ac.cam.cl.dtg.teaching.pottery.worker.Worker;
 
 public class RepoController implements uk.ac.cam.cl.dtg.teaching.pottery.api.RepoController {
 
   protected static final Logger LOG = LoggerFactory.getLogger(RepoController.class);
   private RepoFactory repoFactory;
   private TaskIndex taskIndex;
+  private Worker worker;
 
   /** Create a new RepoController. */
   @Inject
-  public RepoController(RepoFactory repoFactory, TaskIndex taskIndex) {
+  public RepoController(RepoFactory repoFactory, TaskIndex taskIndex, Worker worker) {
     super();
     this.repoFactory = repoFactory;
     this.taskIndex = taskIndex;
+    this.worker = worker;
   }
 
   @Override
   public RepoInfoWithStatus makeRemoteRepo(
       String taskId,
-      Boolean usingTestingVersion,
-      Integer validityMinutes,
+      Boolean usingTestingVersionBoolean,
```

Comment:
Why has the type appeared in the name?

Reply:
So I can have a another variable of a similar name below for finality reasons.